packages/eslint-plugin/tests/rules/no-unnecessary-condition.test.ts

```diff
@@ -545,6 +545,10 @@ type OptionalFoo = Foo | undefined;
 declare const foo: OptionalFoo;
 foo?.[1]?.length;
     `,
+    `
+let variable = 'abc' as string | void;
+variable?.[0];
+    `,
```

Comment:
I looked around the test file and found a few similar tests. Example: `let foo: undefined | { bar: true };`. Could you move this near them?

Reply:
Moved!